api/backups.go

```diff
@@ -0,0 +1,69 @@
+package api
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"encoding/json"
+	"path/filepath"
+	"time"
+
+	"github.com/chadweimer/gomp/fileaccess"
+)
+
+func (h apiHandler) CreateBackup(_ context.Context, _ CreateBackupRequestObject) (CreateBackupResponseObject, error) {
+	// Export recipes
+	exportedRecipes, err := h.db.Backups().ExportRecipes()
+	if err != nil {
+		return nil, err
+	}
+
+	// Export users
+	exportedUsers, err := h.db.Backups().ExportUsers()
+	if err != nil {
+		return nil, err
+	}
+
+	zipData := new(bytes.Buffer)
+	err = fileaccess.CreateZip(zipData, func(writer *zip.Writer) error {
+		// Write the recipes backup to JSON
+		buf, err := json.MarshalIndent(exportedRecipes, "", "  ")
+		if err != nil {
+			return err
+		}
+		if err := fileaccess.WriteFileToZip("recipes.json", bytes.NewBuffer(buf), writer); err != nil {
+			return err
+		}
+
+		// Write the users backup to JSON
+		buf, err = json.MarshalIndent(exportedUsers, "", "  ")
+		if err != nil {
+			return err
+		}
+		if err := fileaccess.WriteFileToZip("users.json", bytes.NewBuffer(buf), writer); err != nil {
```

Comment:
Doing this all in memory is going to be a problem. If there are hundreds of recipes with images, this will likely use a meaningful amount of RAM.

Reply:
Consider hard linking the images on disk to a backup directory, then zipping that.